app/frontend/components/pages/dashboard/dashboardPage.tsx

```diff
@@ -22,201 +22,210 @@ import AccountsDashboard from '../accounts/accountsDashboard'
+  shouldShowNonShelleyCompatibleDialog: boolean
+  displayInfoModal: boolean
+  shouldShowPremiumBanner: boolean
+  shouldShowSaturatedBanner: boolean
+  activeAccountIndex: number
+  shouldNumberAccountsFromOne: boolean
 }
 
-class DashboardMobileContent extends Component<Props, {selectedSubTab}> {
-  constructor(props) {
-    super(props)
-    this.state = {
-      selectedSubTab: 'Transactions',
-    }
-    this.selectSubTab = this.selectSubTab.bind(this)
-  }
-  selectSubTab(name) {
-    if (this.state.selectedSubTab !== name) {
-      this.setState({selectedSubTab: name})
-    }
-  }
-  pages = {
-    'Delegate ADA': DelegatePage,
-    'Current Delegation': CurrentDelegationPage,
-    'Staking history': StakingHistoryPage,
-    'Send ADA': SendAdaPage,
-    'Transactions': TransactionHistory,
-    'Receive ADA': MyAddresses,
-    Keys,
-    'Accounts': AccountsDashboard,
-    'Certificate': PoolOwner,
-  }
-  // TODO: refactor
-  accountsTabs = ['Accounts']
-  stakingTabs = ['Delegate ADA', 'Current Delegation', 'Staking history']
-  sendingTabs = ['Send ADA', 'Transactions', 'Receive ADA']
-  advancedTabs = ['Keys', 'Certificate']
-  render({selectedMainTab}, {selectedSubTab}) {
-    const selectedDefultSubTabs = {
-      Accounts: 'Accounts',
-      Sending: 'Transactions',
-      Staking: 'Delegate ADA',
-      Advanced: 'Keys',
-    }
-    const tabs = {
-      Accounts: this.accountsTabs,
-      Sending: this.sendingTabs,
-      Staking: this.stakingTabs,
-      Advanced: this.advancedTabs,
-    }
-    if (!tabs[selectedMainTab].includes(selectedSubTab)) {
-      this.selectSubTab(selectedDefultSubTabs[selectedMainTab])
-    }
-    const Page = this.pages[selectedSubTab]
-    return (
-      <div className="dashboard-content">
-        <ul className="dashboard-tabs">
-          {tabs[selectedMainTab].map((name, i) => (
-            <SubTab
+const DashboardPage = ({
+  selectedMainTab,
+  displayInfoModal,
+  isShelleyCompatible,
+  shouldShowNonShelleyCompatibleDialog,
+  shouldShowPremiumBanner,
+  shouldShowSaturatedBanner,
+  activeAccountIndex,
+  shouldNumberAccountsFromOne,
+  selectMainTab,
+}: Props) => {
+  return (
+    <div className="page-wrapper">
+      <ErrorModals />
+      {isShelleyCompatible && displayInfoModal && <InfoModal />}
+      {shouldShowNonShelleyCompatibleDialog && <NotShelleyCompatibleDialog />}
+      {!isShelleyCompatible && <DashboardErrorBanner />}
+      {shouldShowPremiumBanner && <PremiumBanner />}
+      {shouldShowSaturatedBanner && <SaturationErrorBanner />}
+      {ADALITE_CONFIG.ADALITE_CARDANO_VERSION === 'shelley' && (
+        <ul className="tabinator">
+          {Object.values(MainTabs).map((name, i) => (
+            <MainTab
               key={i}
               name={name}
-              selectedTab={selectedSubTab}
-              selectTab={this.selectSubTab}
+              isSelected={name === selectedMainTab}
+              selectTab={() => selectMainTab(name)}
```

Comment:
shouldn't this be just `selectMainTab(name)` instead of `() => selectMainTab(name)` due to `(name: MainTabs) => void` type?

Reply:
You are partly right, In a way, what I need is to pass there a function, either `() => selectMainTab(name)` or `selectMainTab` but the second one the better option since it agrees with the type. Thnx.